app.js

```diff
@@ -1,36 +1,144 @@
-/**
- * Copyright 2017, Google, Inc.
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 'use strict';
 
-// [START gae_node_request_example]
 const express = require('express');
+const cookieParser = require('cookie-parser');
+const uniqueString = require('unique-string');
+
+const logger = require('./logger');
+
+// TODO: none of this setup is pretty
+const {CloudStorage, MemoryStorage} = require('./storage');
+const storage = process.env.NODE_ENV === 'production'
+   ? new CloudStorage
+   : new MemoryStorage;
+
+const secrets = process.env.NODE_ENV === 'test'
+    ? require('./secrets.sample.json')
+    : require('./secrets.json');
+
+const github = require('./github')({
+  auth: `token ${secrets.github.token}`,
+});
+
+const Tests = require('./tests');
+const tests = new Tests({
+  manifest: require('./MANIFEST.json'),
+  host: process.env.NODE_ENV === 'production'
+      ? 'mdn-bcd-collector.appspot.com'
+      : 'localhost:8080',
+  httpOnly: process.env.NODE_ENV !== 'production',
+});
+
+function cookieSession(req, res, next) {
+  req.sessionID = req.cookies.sid;
+  if (!req.sessionID) {
+    req.sessionID = uniqueString();
+    res.cookie('sid', req.sessionID);
+  }
+  next();
+}
 
 const app = express();
+app.use(cookieParser());
+app.use(cookieSession);
+app.use(express.json({limit: '32mb'}));
+app.use(express.static('static'));
+app.use(express.static('generated'));
 
-app.get('/', (req, res) => {
-  res
-    .status(200)
-    .send('Hello, world!')
-    .end();
+app.get('/api/tests', (req, res) => {
+  const {after, limit} = req.query;
+  const list = tests.list(after, limit ? +limit : 0);
```

Comment:
Is the plus sign (`+limit`) just for coercing `limit` into int?

Reply:
Yes, this is because of the `afterIndex + 1 + limit` expression in the called function. `1 + '1'` is the string "11" in JavaScript, so I needed to coerce to a number somewhere.